test/plug/adapters/cowboy/conn_test.exs

```diff
@@ -129,6 +135,34 @@ defmodule Plug.Adapters.Cowboy.ConnTest do
            {"x-sample", "value"}
   end
 
+  test "allows customized statuses based on config" do
+    assert {500, _headers, _} = request :get, "/send_451"
+
+    Application.put_env(:plug, :statuses, %{451 => "Unavailable For Legal Reasons"})
```

Comment:
It is OK to set this in Plug's `config/config.exs` so we don't need to recompile the file during tests.